### DISCUSSION TITLE
Question about creating a microstructure where only some grains have lamella

### DISCUSSION THREAD

**Discussion by mauroarcidiacono** [neperfepx/neper#829] [Q&A] [answered]
Hi,

I hope all is well. I am creating a microstructural model with the following commands:
```
neper -T -n 100 -domain "square(10,10)" -dim 2 -id 4 -morpho "gg" -reg 1 -oridescriptor euler-bunge -format tess -o polycrystal_basic
neper -V polycrystal_basic.tess -datacellcol id -datacelltrs 0.7 -print polycrystal_basic_image
neper -M polycrystal_basic.tess -elttype tri -statelset "area" -order 1 -rcl 1 -meshqualexpr Odis -meshqualmin 1 -format inp,msh -o polycrystal_basic
```
Then, with a Python script I evaluate the combination of areas which leads to the closest value to an input area. With that information, I proceed to create the lamwidth.txt and the lamnormal.txt files. The issue I have is that I only want some grains to have lamella, not all of them. I also would like to know which are those grains. 

This are the text files that I am creating:
[lamwidth.txt](https://github.com/neperfepx/neper/files/14637680/lamwidth.txt)
[lamnormal.txt](https://github.com/neperfepx/neper/files/14637681/lamnormal.txt)

I am trying to "trick" Neper by using very large lam relationship values and I am assigning 1 0 0 as orientation for the grains that should not have lamella. Then, I use the following commands:
```
neper -T -n 100::from_morpho -domain "square(10,10)" -dim 2 -id 4::1 -morpho "gg::lamellar(w=msfile(lamwidth.txt),v=msfile(lamnormal.txt),pos=optimal)" -group lam -reg 1 -rsel 0.1 -oridescriptor euler-bunge -group lam -statcell "scaleid(1)" -format tess,ori -o polycrystal_laminated
neper -V polycrystal_laminated.tess -datacellcol id -datacelltrs 0.7 -print polycrystal_laminated_image
```

Which results in this image:
![image](https://github.com/neperfepx/neper/assets/76496290/c6d82cab-b5d9-43f0-80e0-72bdb965c567)

The problem I have is that some laths are thicker than others. Am I following the right approach to solve this problem? Do you have a recommendation on how to solve it?

Thank you very much for your help.

**Comment by rquey** [comment, marked as answer]
The different lamella widths come from the fact that the normals provided in `lamnormal.txt` are out of plane. For a 2D tess, they should be in-plane to get the expected widths.

Here is what I get with in-plane normals:
```
awk '{print $1,$2/sqrt($2*$2+$3*$3),$3/sqrt($2*$2+$3*$3),0}' lamnormal.txt > lamnormal.txt2
neper -T -n 100::from_morpho -domain "square(10,10)" -dim 2 -id 4::1 -morpho "gg::lamellar(w=msfile(lamwidth.txt),v=msfile(lamnormal.txt2),pos=optimal)" -group lam -reg 1 -rsel 0.1 -oridescriptor euler-bunge -group lam -statcell "scaleid(1)" -format tess,ori -o polycrystal_laminated
 ```
![polycrystal_laminated_image](https://github.com/neperfepx/neper/assets/2659376/b302e1e1-4adc-47b8-9bc3-c6b62a585ddd)

And instead of the "trick", you could use a [multiscale cell file](https://neper.info/doc/fileformat.html#multiscale-cell-file) (although it will produce the same result...).
